packages/web/src/js/Modal.ts

```diff
@@ -48,11 +50,17 @@ class Modal extends BaseComponent {
   addEventListeners() {
     EventHandler.on(this.element, 'close', (event: KeyboardEvent) => this.onDialogClose(event));
     EventHandler.on(window, 'click', (event: Event & { target: Window }) => this.onClick(event));
+    if (this.isSafari) {
```

Comment:
Maybe this is not necessary. I have just looked at how this solves [Bootstrap](https://github.com/twbs/bootstrap/blob/5036f70e1c80cebb270aca8c1b422f8d4d4bd47f/js/src/modal.js#L224) and they are using `mousdown` event (https://developer.mozilla.org/en-US/docs/Web/API/Element/mousedown_event) instead of the `click`.

Please give this a try.

Reply:
I am not sure about this. I have tried to deep dive through https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html and https://sitr.us/2011/07/28/how-mobile-safari-emulates-mouse-events.html. Also tried to test this on browserstack.com. I think the difference between our approach and Bootstraps's is in using the dialog element and the `::backdrop` pseudoelement.

Anyway. I have tested this on my iPhone 13 and Safari and it is working, but on my iPad and @adamkudrna tried it also, but this solution does not work.

I have also tried different browsers on both iPhone and iPad like Firefox or Chrome and in those browsers, it is working without any modification, eg. tried on the main branch.

So there must be difference between handling those events on iPhone and iPad in Safari 🤔

Reply:
On Apple devices, the Safari core is always used for all browsers, whatever the name of the browser